lib/iris/fileformats/netcdf.py

```diff
@@ -1818,29 +2051,149 @@ def _get_coord_variable_name(self, cube, coord):
+                if cube.coords(coord):
+                    # It is a regular cube coordinate.
+                    # Auto-generate a name based on the dims.
+                    name = ""
+                    for dim in cube.coord_dims(coord):
+                        name += "dim{}".format(dim)
+                    # Handle scalar coordinate (dims == ()).
+                    if not name:
+                        name = "unknown_scalar"
+                else:
+                    # Not a cube coord, so must be a connectivity or
+                    # element-coordinate of the mesh.
+                    # Name it for it's first dim, i.e. mesh-dim of its location.
+                    from iris.experimental.ugrid import Connectivity
+
+                    if isinstance(coord, Connectivity):
+                        location = coord.cf_role.split("_")[0]
+                    else:
+                        # Must be a mesh-element coordinate.
+                        location = None
+                        for test_location in MESH_LOCATIONS:
+                            include_key = f"include_{test_location}s"
+                            if coord in cube.mesh.coords({include_key: True}):
+                                location = test_location
+                                break
+                        assert location is not None
+                    location_dim_attr = f"{location}_dimension"
+                    name = getattr(cube.mesh, location_dim_attr)
+
             # Convert to lower case and replace whitespace by underscores.
             cf_name = "_".join(name.lower().split())
 
         cf_name = self.cf_valid_var_name(cf_name)
         return cf_name
 
-    _ELEMENT_TYPE_NAMES = {
-        iris.coords.DimCoord: "coordinate",
-        iris.coords.AuxCoord: "coordinate",
-        iris.coords.CellMeasure: "cell-measure",
-        iris.coords.AncillaryVariable: "ancillary-variable",
-    }
+    def _get_mesh_variable_name(self, mesh):
+        """
+        Returns a CF-netCDF variable name for the mesh.
+
+        Args:
+
+        * mesh (:class:`iris.experimental.ugrid.Mesh`):
+            An instance of a Mesh for which a CF-netCDF variable name is
+            required.
+
+        Returns:
+            A CF-netCDF variable name as a string.
+
+        """
+        cf_name = mesh.var_name
+        if mesh.var_name is None:
+            # Prefer var-name, but accept long/standard as aliases.
+            cf_name = mesh.var_name or mesh.standard_name or mesh.long_name
+            if not cf_name:
+                # Auto-generate a name based on mesh properties.
+                cf_name = f"Mesh_{mesh.topology_dimension}d"
+
+        # Ensure valid form for var-name.
+        cf_name = self.cf_valid_var_name(cf_name)
+        return cf_name
 
-    def _create_generic_cf_array_var(self, cube, cube_dim_names, element):
+    def _create_mesh(self, cube, mesh):
```

Comment:
@pp-mo The `cube` argument is not used within `_create_mesh`

Reply:
:+1: good spot
Should have noticed PyCharm telling me this.